server/src/main/java/org/cloudfoundry/identity/uaa/provider/token/JwtBearerAssertionTokenAuthenticator.java

```diff
@@ -121,14 +131,58 @@ private Authentication authenticateJwtAssertionToken(final Jwt jwt, String devic
         throw new BadCredentialsException("Authentication of client failed.");
     }
 
+    private void validateClaims(final Map<String, Object> claims) {
+        // require subject claim
+        if (!StringUtils.hasText((String) claims.get(ClaimConstants.SUB))) {
+            throw new InvalidTokenException("Subject claim is required in jwt-bearer assertion.");
+        }
+
+        // require tenant_id claim
+        if (!StringUtils.hasText((String) claims.get(ClaimConstants.TENANT_ID))) {
+            throw new InvalidTokenException("Tenant id claim is required in jwt-bearer assertion.");
+        }
+
+        // require issuer claim
+        if (!StringUtils.hasText((String) claims.get(ClaimConstants.ISS))) {
+            throw new InvalidTokenException("Issuer claim is required in jwt-bearer assertion.");
```

Comment:
for validation errors, pls use the actual claim names in the errors. 'iss' in this case.